### PR TITLE
Add additional parameters for webapp site config

### LINKED ISSUE #421: azure_rm_webapp: Client affinity is not disabled
##### SUMMARY
Setting client_affinity_enabled to false does not disable client affinity.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
azure_rm_webapp

##### ANSIBLE VERSION
```ansible 2.10.2
  config file = None
  configured module search path = ['/home/takuya/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/takuya/ansible2.10/lib/python3.8/site-packages/ansible
  executable location = /home/takuya/ansible2.10/bin/ansible
  python version = 3.8.5 (default, Jul 28 2020, 12:59:40) [GCC 9.3.0]
```

##### OS / ENVIRONMENT
Ubuntu 20.04.1 LTS

##### STEPS TO REPRODUCE

```yaml
- azure_rm_webapp:
    resource_group: myResourceGroup
    name: myWinWebapp
    plan: myAppServicePlan
    client_affinity_enabled: no
```

##### EXPECTED RESULTS
Webapps will be created with client affinity disabled.

##### ACTUAL RESULTS
Client affinity is enabled.

### PR DESCRIPTION
**Pull request by l3ender**
##### SUMMARY
This PR updates the `azure_rm_webapp` module to support the following parameters:

* `always_on`: Keeps the app loaded even when there's no traffic.
* `min_tls_version`: The minimum TLS encryption version required for the app.
* `ftps_state`: The state of the FTP/FTPS service.

This PR also has a small change which fixes https://github.com/ansible-collections/azure/issues/421.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
azure_rm_webapp
azure_rm_webapp_info

##### ADDITIONAL INFORMATION

Please see updated test case for detail and validation.

### CODE REVIEW

**Comment by l3ender** [comment]
@Fred-sun Another PR--thank you! ;)

**Comment by Fred-sun** [comment]
> @Fred-sun Another PR--thank you! ;)

@l3ender Okay, we'll move forward with the merger as soon as possible. Thank you for your contribution!

File: plugins/modules/azure_rm_webapp.py [hunk 1]
```diff
@@ -662,7 +699,7 @@ def exec_module(self, **kwargs):
             if self.https_only is not None:
                 self.site.https_only = self.https_only
 
-            if self.client_affinity_enabled:
+            if self.client_affinity_enabled is not None:
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_webapp.py]: If you pass the value directly（self.site.client_affinity_enabled = self.client_affinity_enabled）, you can do it without if. Thank you very much!
```suggestion

```

**Comment by Fred-sun** [comment]
@l3ender Any updates on the comments above? Thank you very much!

**Comment by l3ender** [comment]
Hi @Fred-sun, I don't see any comments. Can you let me know what you'd like me to do?

Thanks!

**Comment by Fred-sun** [comment]
@l3ender The client_affinity_enabled parameter is a bool and the default value is True. You are advised not to change this parameter. Thank you very much!

```e.g.
line 702:
            if self.client_affinity_enabled is not None:
          rebase  to ---  if self.client_affinity_enabled:

**Comment by l3ender** [comment]
> The client_affinity_enabled parameter is a bool and the default value is True.

The change on [line 702 of the azure_rm_webapp.py file](https://github.com/ansible-collections/azure/pull/695/files#diff-8fba5954c1d4f1734bdafec38891ba3b6c93fe82a6f34ffb632507a637cbf9b4R702) is actually a bugfix where `client_affinity_enabled` is not correctly configured when module input parameter is false. There is an issue logged for it at https://github.com/ansible-collections/azure/issues/421.

I noticed the same issue with `client_affinity_enabled` incorrect behavior when adding the new parameters in this PR, so I went ahead and fixed it. My apologies for combining the changes into a single PR and for not providing any detail on the reason for that change. I hope it is okay!!

Thank you!

File: plugins/modules/azure_rm_webapp.py [hunk 2]
```diff
@@ -662,7 +699,6 @@ def exec_module(self, **kwargs):
             if self.https_only is not None:
                 self.site.https_only = self.https_only
 
-            if self.client_affinity_enabled:
                 self.site.client_affinity_enabled = self.client_affinity_enabled
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_webapp.py]: Indent should also be cancelled, equal to if!

**Comment by Fred-sun** [comment]
@l3ender Thanks for you contribute!

**Comment by l3ender** [comment]
Thank you for the review and suggestions. I have updated and please let me know if there is anything else.

**Comment by xuzhang3** [comment]
LGTM, shipping it.

**Pull request merged**